File: ContainerWithMostWater/ContainerWithMostWater.py
def maxArea(height):
    answer = 0
    # CRIO_SOLUTION_START_MODULE_CONTAINER_WITH_MOST_WATER

    # CRIO_SOLUTION_END_MODULE_CONTAINER_WITH_MOST_WATER
    return answer

if __name__ == '__main__':
    n = int(input())
    height = input().split()
    height = [int(i) for i in height]
    result = maxArea(height)
    print(result)
    

File: ContainerWithMostWater/ContainerWithMostWater.cpp
#include "../crio/cpp/io/FastIO.hpp"
#include "../crio/cpp/io/PrintMatrix.hpp"
#include "../crio/cpp/io/ReadMatrix.hpp"
#include <bits/stdc++.h>
using namespace std;

class ContainerWithMostWater {
  public:
    
    int maxArea(vector<int> &nums)
{
    int ans = INT_MIN;
    for(int i = 0; i < (int)nums.size()-1; i++)
    {
        for(int j = i+1; j < (int)nums.size(); j++)
        {
            int curr = min(nums[i], nums[j])*(j-i);
            ans = max(ans, curr);
        }
    }
    return ans;
}
};

int main() {
    FastIO();
    int n;
    cin >> n;
    vector<int> height;
    ReadMatrix<int>().OneDMatrix(n, height);
    int result = ContainerWithMostWater().maxArea(height);
    cout << result;
    return 0;
}
